gridstatus/ieso.py

```diff
@@ -0,0 +1,517 @@
+# and only covers the current date.
+LOAD_FORECAST_URL = (
+    "https://www.ieso.ca/-/media/Files/IESO/Power-Data/Ontario-Demand-multiday.ashx"
+)
+MAXIMUM_DAYS_IN_PAST_FOR_LOAD_FORECAST = 5
+MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST = 1
+
+"""ZONAL LOAD FORECAST CONSTANTS"""
+ZONAL_LOAD_FORECAST_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand"
+
+# Each forecast file contains data from the day in the filename going forward for
+# 34 days. The most recent file does not have a date in the filename.
+ZONAL_LOAD_FORECAST_TEMPLATE_URL = (
+    f"{ZONAL_LOAD_FORECAST_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_ZONAL_LOAD_FORECAST = 90
+# The farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_ZONAL_LOAD_FORECAST = 34
+
+
+MINUTES_INTERVAL = 5
+MINUTES_INTERVALS_IN_AN_HOUR = 12
+HOUR_INTERVAL = 1
+
+# Default namespace used in the XML files
+NAMESPACES_FOR_XML = {"": "http://www.ieso.ca/schema"}
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
+
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get 5-minute load for the Market and Ontario for a given date or from
+        date to end date.
+
+        Args:
+            date (datetime.date | datetime.datetime | str): The date to get the load for
+                Can be a `datetime.date` or `datetime.datetime` object, or a string
+                with the values "today" or "latest". If `end` is None, returns
+                only data for this date.
+            end (datetime.date | datetime.datetime, optional): End date. Defaults None
+                If provided, returns data from `date` to `end` date. The `end` can be a
+                `datetime.date` or `datetime.datetime` object.
+            verbose (bool, optional): Print verbose output. Defaults to False.
+            frequency (str, optional): Frequency of data. Defaults to "5min".
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        # We have to add special time handling here because the existing utils
+        # don't perfectly handle the 1 hour data files.
+        if isinstance(date, tuple):
```

Comment:
why cant `@support_date_range(frequency="HOUR_START")` decorate `get_load`? if it does, would it handle all of this?

Reply:
The decorator does not correctly handle "today" for hourly data. If I use the decorator and pass in "today", the function only runs once for the first hour of the day.

Reply:
ahh. i think it might be wise to try to update the decorator to handle that case correctly. how messy would that be?

Reply:
It shouldn't be too bad since the "HOUR_START" frequency is only used in two other places.